upup/models/cloudup/resources/addons/networking.romana/k8s-1.6.yaml.template

```diff
@@ -0,0 +1,312 @@
+      containers:
+      - name: romana-agent
+        image: quay.io/romana/agent:v2.0-preview.2
+        imagePullPolicy: Always
+        env:
+        - name: NODENAME
+          valueFrom:
+            fieldRef:
+              fieldPath: spec.nodeName
+        args:
+        - --service-cluster-ip-range={{ .ServiceClusterIPRange }}
+        securityContext:
+          privileged: true
+        volumeMounts:
+        - name: host-usr-local-bin
+          mountPath: /host/usr/local/bin
+        - name: host-etc-romana
+          mountPath: /host/etc/romana
+        - name: host-cni-bin
+          mountPath: /host/opt/cni/bin
+        - name: host-cni-net-d
+          mountPath: /host/etc/cni/net.d
+        - name: run-path
+          mountPath: /var/run/romana
+      volumes:
+      - name: host-usr-local-bin
+        hostPath:
+          path: /usr/local/bin
+      - name: host-etc-romana
+        hostPath:
+          path: /etc/romana
+      - name: host-cni-bin
+        hostPath:
+          path: /opt/cni/bin
+      - name: host-cni-net-d
+        hostPath:
+          path: /etc/cni/net.d
+      - name: run-path
+        hostPath:
+          path: /var/run/romana
+---
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1beta1
+metadata:
+  name: romana-aws
+rules:
+- apiGroups:
+  - "*"
+  resources:
+  - nodes
+  verbs:
+  - get
+  - list
+  - watch
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: romana-aws
+  namespace: kube-system
+---
+kind: ClusterRoleBinding
+apiVersion: rbac.authorization.k8s.io/v1beta1
+metadata:
+  name: romana-aws
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: romana-aws
+subjects:
+- kind: ServiceAccount
+  name: romana-aws
+  namespace: kube-system
+---
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: romana-aws
+  namespace: kube-system
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        romana-app: aws
+    spec:
+      nodeSelector:
+        node-role.kubernetes.io/master: ""
+      hostNetwork: true
+      serviceAccountName: romana-aws
+      tolerations:
+      - key: node-role.kubernetes.io/master
+        effect: NoSchedule
+      containers:
+      - name: romana-aws
+        image: quay.io/romana/aws:v2.0-preview.2
+        imagePullPolicy: Always
+---
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: romana-vpcrouter
+  namespace: kube-system
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        romana-app: vpcrouter
+    spec:
+      nodeSelector:
+        node-role.kubernetes.io/master: ""
+      hostNetwork: true
+      tolerations:
+      - key: node-role.kubernetes.io/master
+        effect: NoSchedule
+      containers:
+      - name: romana-vpcrouter
+        image: quay.io/romana/vpcrouter-romana-plugin
+        imagePullPolicy: Always
+        args:
+        - --etcd_use_v2
+        - --etcd_addr={{ .Networking.Romana.EtcdServiceIP }}
```

Comment:
How is calico setting this? Can we do hostname(s) instead of IP address?

Reply:
This component, `romana-vpcrouter` is involved in creating route entries so that pods can be reached, and runs on the host network instead of as a plain pod.
We'd originally configured it as a regular pod and using a DNS address for etcd, but it didn't work. It provides the routing to pods, such as kube-dns. When the routes don't exist yet, it can't find etcd to get the information to add those routes.

For those reasons, it uses hostNetworking and a predeclared service ip for the etcd address.